src/StateFile.cxx

```diff
@@ -118,11 +124,16 @@ try {
 
 	const char *line;
 	while ((line = file.ReadLine()) != nullptr) {
-		success = read_sw_volume_state(line, partition.outputs) ||
-			audio_output_state_read(line, partition.outputs) ||
-			playlist_state_restore(line, file, song_loader,
+		success =            read_sw_volume_state(line, partition.outputs);
+		success = success || audio_output_state_read(line, partition.outputs);
+		success = success || playlist_state_restore(line, file, song_loader,
 					       partition.playlist,
 					       partition.pc);
+#ifdef ENABLE_DATABASE
+		success = success || ((CompositeStorage *)partition.instance.storage)->MountStateRestore(
```

Comment:
This needs a short-circuiting `||` operator.

Reply:
There is a short-circuiting `||` operator. Am I missing something?

Reply:
Sorry, you're right.